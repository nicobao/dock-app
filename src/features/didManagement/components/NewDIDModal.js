import {Modal} from '../../../components/Modal';
import React from 'react';
import {Stack} from 'native-base';
import {
  DownloadIcon,
  OptionList,
  PlusCircleIcon,
  Theme,
  Typography,
} from '../../../design-system';
import {translate} from '../../../locales';
import {addTestId} from '../../../core/automation-utils';
import {navigate} from '../../../core/navigation';
import {Routes} from '../../../core/routes';

export function NewDIDModal({visible, onClose, onImportDID}) {
  const content = (
    <Stack p={8} testID="addDIDModal">
      <Typography variant="h1">{translate('didManagement.add_did')}</Typography>
      <OptionList
        mt={5}
        items={[
          {
            testID: addTestId('CreateNewDID').testID,
            title: translate('didManagement.create_new_did'),
            icon: (
              <PlusCircleIcon
                style={{
                  color: Theme.colors.secondaryIconColor,
                }}
              />
            ),
            onPress: () => {
              onClose();
              navigate(Routes.DID_MANAGEMENT_NEW_DID);
            },
          },
          {
            testID: addTestId('ImportExistingDIDBtn').testID,
            title: translate('didManagement.import_existing_did'),
            icon: (
              <DownloadIcon
                style={{
                  color: Theme.colors.secondaryIconColor,
                }}
              />
            ),
            onPress: onImportDID,
          },
        ]}
      />
    </Stack>
  );
  return (
    <Modal visible={visible} onClose={onClose} onBackButtonPress={onClose}>
      {content}
    </Modal>
  );
}
