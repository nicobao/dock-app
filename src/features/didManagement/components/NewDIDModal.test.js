import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {NewDIDModal} from './NewDIDModal';

const mockStore = configureMockStore();

describe('NewDIDModal', () => {
  it('should render correctly for modal is visible', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      <NewDIDModal
        onImportDID={jest.fn()}
        visible={true}
        onClose={jest.fn()}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('should render correctly for modal is not visible', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      <NewDIDModal
        onImportDID={jest.fn()}
        visible={false}
        onClose={jest.fn()}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
