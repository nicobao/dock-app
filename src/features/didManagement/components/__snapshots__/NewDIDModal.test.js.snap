// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewDIDModal should render correctly for modal is not visible 1`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.7}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  coverScreen={true}
  customBackdrop={null}
  deviceHeight={null}
  deviceWidth={null}
  hasBackdrop={true}
  hideModalContentWhileAnimating={false}
  isVisible={false}
  onBackButtonPress={[MockFunction]}
  onBackdropPress={[MockFunction]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onSwipeComplete={[MockFunction]}
  panResponderThreshold={4}
  propagateSwipe={false}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  style={
    Object {
      "justifyContent": "flex-end",
      "margin": 0,
    }
  }
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeDirection={
    Array [
      "down",
    ]
  }
  swipeThreshold={100}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "backgroundColor": "#18181B",
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 667,
        "width": "100%",
      }
    }
  >
    <Memo(ForwardRef(Stack))
      alignItems="center"
      mt={6}
    >
      <Memo(ForwardRef(Box))
        backgroundColor="#3F3F46"
        borderRadius={6}
        height="4px"
        width="36px"
      />
    </Memo(ForwardRef(Stack))>
    <Memo(ForwardRef(Stack))
      p={8}
      testID="addDIDModal"
    >
      <Styled(Component)
        variant="h1"
      >
        Add DID
      </Styled(Component)>
      <OptionList
        items={
          Array [
            Object {
              "icon": <SvgFile
                style={
                  Object {
                    "color": "#ffffff",
                  }
                }
              />,
              "onPress": [Function],
              "testID": "CreateNewDID",
              "title": "Create New DID",
            },
            Object {
              "icon": <SvgFile
                style={
                  Object {
                    "color": "#ffffff",
                  }
                }
              />,
              "onPress": [Function],
              "testID": "ImportExistingDIDBtn",
              "title": "Import existing DID",
            },
          ]
        }
        mt={5}
      />
    </Memo(ForwardRef(Stack))>
  </View>
</ReactNativeModal>
`;

exports[`NewDIDModal should render correctly for modal is visible 1`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.7}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  coverScreen={true}
  customBackdrop={null}
  deviceHeight={null}
  deviceWidth={null}
  hasBackdrop={true}
  hideModalContentWhileAnimating={false}
  isVisible={true}
  onBackButtonPress={[MockFunction]}
  onBackdropPress={[MockFunction]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onSwipeComplete={[MockFunction]}
  panResponderThreshold={4}
  propagateSwipe={false}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  style={
    Object {
      "justifyContent": "flex-end",
      "margin": 0,
    }
  }
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeDirection={
    Array [
      "down",
    ]
  }
  swipeThreshold={100}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "backgroundColor": "#18181B",
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 667,
        "width": "100%",
      }
    }
  >
    <Memo(ForwardRef(Stack))
      alignItems="center"
      mt={6}
    >
      <Memo(ForwardRef(Box))
        backgroundColor="#3F3F46"
        borderRadius={6}
        height="4px"
        width="36px"
      />
    </Memo(ForwardRef(Stack))>
    <Memo(ForwardRef(Stack))
      p={8}
      testID="addDIDModal"
    >
      <Styled(Component)
        variant="h1"
      >
        Add DID
      </Styled(Component)>
      <OptionList
        items={
          Array [
            Object {
              "icon": <SvgFile
                style={
                  Object {
                    "color": "#ffffff",
                  }
                }
              />,
              "onPress": [Function],
              "testID": "CreateNewDID",
              "title": "Create New DID",
            },
            Object {
              "icon": <SvgFile
                style={
                  Object {
                    "color": "#ffffff",
                  }
                }
              />,
              "onPress": [Function],
              "testID": "ImportExistingDIDBtn",
              "title": "Import existing DID",
            },
          ]
        }
        mt={5}
      />
    </Memo(ForwardRef(Stack))>
  </View>
</ReactNativeModal>
`;
